src/clients/graphql/storefront_client.ts

```diff
@@ -0,0 +1,17 @@
+import {Context} from '../../context';
+import {ShopifyHeader} from '../../base_types';
+
+import {GraphqlClient, AccessTokenHeader} from './graphql_client';
+
+export class StorefrontClient extends GraphqlClient {
+  protected baseApiPath = '/api';
+
+  protected getAccessTokenHeader(): AccessTokenHeader {
+    return {
+      header: ShopifyHeader.StorefrontAccessToken,
+      value: (
+        Context.IS_PRIVATE_APP ? Context.PRIVATE_APP_STOREFRONT_ACCESS_TOKEN || this.accessToken : this.accessToken
```

Comment:
I'm not sure I understand the `||` in the first condition on this line. Can you explain what's happening here?

Reply:
We're basically defaulting to the given access token if the setting isn't there. Not sure we necessarily want to do that, though - we don't in the GraphQL client, but then again it uses the secret key which is mandatory.

Reply:
Does
```
Context.PRIVATE_APP_STOREFRONT_ACCESS_TOKEN || this.accessToken
```
evaluate to a string or a boolean?

Reply:
String:
```
a = 'a';
"a"
b = 'b';
"b"
a || b
"a"
```